csrc/ops/arith.cpp

```diff
@@ -447,14 +447,28 @@ TensorView* eye(Val* size, DataType dtype) {
 NVFUSER_DEFINE_UNARY_OP(ceil, Ceil)
 NVFUSER_DEFINE_UNARY_OP(floor, Floor)
 NVFUSER_DEFINE_UNARY_OP(frac, Frac)
-NVFUSER_DEFINE_UNARY_OP(neg, Neg)
 NVFUSER_DEFINE_UNARY_OP(relu, Relu)
 NVFUSER_DEFINE_UNARY_OP(round, Round)
 NVFUSER_DEFINE_UNARY_OP(silu, Silu)
 NVFUSER_DEFINE_UNARY_OP(trunc, Trunc)
 NVFUSER_DEFINE_UNARY_OP(print, Print)
 #undef NVFUSER_DEFINE_UNARY_OP
 
+// Promote half to single for neg. Eventually, `neg` should probably be defined
+// using NVFUSER_DEFINE_UNARY_OP. However, currently, nvFuser codegen misses
+// certain header files for half types and therefore has no access to data types
+// like `__nv_bfloat16`  and intrinsics like `__hneg`. So, as a workaround to
+// #1541, we promote half types to single.
+//
+// Caveat: for `neg`, don't promote integer types to single.
```

Comment:
nitpick: this reads like we *are* promoting integer to single... but we are not, right?!

Reply:
I rewrote the comment. I hope it's more clear now.